server/ai_live_video.go

```diff
@@ -355,6 +363,29 @@ func startTrickleSubscribe(ctx context.Context, url *url.URL, params aiRequestPa
 			clog.V(8).Info(ctx, "trickle subscribe read data completed", "seq", seq, "bytes", humanize.Bytes(uint64(n)), "took", time.Since(copyStartTime))
 		}
 	}()
+
+	// watchdog: fires if orch does not produce segments for too long
+	go func() {
+		for {
+			select {
+			case <-segmentTicker.C:
+				// check whether this timeout is due to missing input
+				params.liveParams.mu.Lock()
+				lastInputSegmentTime := params.liveParams.lastSegmentTime
+				params.liveParams.mu.Unlock()
+				lastInputSegmentAge := time.Since(lastInputSegmentTime)
+				hasRecentInput := lastInputSegmentAge < segmentTimeout/2
+				if hasRecentInput && params.inputStreamExists() {
```

Comment:
Shouldn't it be `!hasRecentInput`? I think we should not suspend if there is a recent input segment.